provider_test.go

```diff
@@ -54,18 +70,66 @@ func TestRetry(test *testing.T) {
 	}
 }
 
-func TestGetAPI(test *testing.T) {
+func TestGetRegistryAPI(test *testing.T) {
 	// when the address is an empty string, we get a nullAPI
 	var api client.API
 
-	api, err := getAPI("", 1)
+	api, err := getAPI(Conf{
+		ClientDriver : "etcd",
+		Endpoint : "",
+		EtcdKeyPrefix : "",
+		KeyFile : "",
+		CertFile : "",
+		CAFile : "",
+		Tunnel : "",
+		KnownHostsFile : "",
+		SSHUserName : "",
+
+		StrictHostKeyChecking : false,
+
+		SSHTimeout : 1.0,
+		RequestTimeout : 1.0,
+	})
+
+	if err != nil {
+		test.Fatal(err)
+	}
+
+	switch api.(type) {
+	case *client.RegistryClient:
+		// pass!
+	default:
+		test.Errorf("didn't get nullAPI, got %s instead", api)
+	}
+}
+
+func TestGetHTTPAPI(test *testing.T) {
+	// when the address is an empty string, we get a nullAPI
+	var api client.API
+
+	api, err := getAPI(Conf{
+		ClientDriver : "api",
+		Endpoint : "http://falselocation.test:8080",
+		EtcdKeyPrefix : "",
+		KeyFile : "",
+		CertFile : "",
+		CAFile : "",
+		Tunnel : "",
+		KnownHostsFile : "",
+		SSHUserName : "",
+
+		StrictHostKeyChecking : false,
+
+		SSHTimeout : 1.0,
+		RequestTimeout : 1.0,
+	})
 
 	if err != nil {
 		test.Fatal(err)
 	}
 
 	switch api.(type) {
-	case nullAPI:
+	case *client.HTTPClient:
```

Comment:
This seems contrary to the description of the test, "when the address is an empty string, we get a nullAPI", and the error message below, "didn't get nullAPI, got %s instead"